### PR TITLE
Rewarded Video Ads

### PR DESCRIPTION
**Pull request by NicholasSmilovic**
First PR from a massive noob here but I would love to have this feature. Android code references https://github.com/4devcoding/nativescript-admob and I attempted to do the iOS portion, but again I have no idea what I'm doing. 

If someone is willing to review and tell me what I'm going wrong with delegates I'll do my best to clean this up.

https://developers.google.com/admob/ios/rewarded-video
https://developers.google.com/ad-manager/mobile-ads-sdk/ios/api/reference/Classes/GADRewardBasedVideoAd

### CODE REVIEW

File: src/admob.ios.js [hunk 1]
```diff
@@ -234,5 +234,100 @@ admob.hideBanner = function () {
...
+
+admob.preloadRewardedVideoAd = function (arg) {
+  return new Promise(function (resolve, reject) {
+    try {
+      admob.videoView = GADRewardBasedVideoAd.sharedInstance();
+
+      // Rewarded ads must be loaded before they can be shown, so adding a listener
+      console.log("plz")
+      var delegate = GADRewardBasedVideoAdDelegateImpl.new().initWithCallback(function (ad, error) {
+        console.log("********---", ad)
```
Comment (NicholasSmilovic) [inline src/admob.ios.js]: Whats holding me back is getting into this function.

**Comment by NicholasSmilovic** [comment]
iOS should be working now.

**Comment by Roar1827** [comment]
Nice work @NicholasSmilovic !

Any chance anyone can help get this across the finish line? 😃

@JCenas @EddyVerbruggen @sedobrengocce @muratcorlu @cloudlegend @svzi

File: src/admob.ios.js [hunk 2]
```diff
@@ -285,17 +285,18 @@ var GADRewardBasedVideoAdDelegateImpl = (function (_super) {
...
 let rewardedVideoCallbacks = {
-  onRewarded: () => {},
+  onRewarded: () => { console.warn("onRewarded callback not set")},
   onRewardedVideoAdLeftApplication: () => {},
   onRewardedVideoAdClosed: () => {},
   onRewardedVideoAdOpened: () => {},
   onRewardedVideoStarted: () => {},
   onRewardedVideoCompleted: () => {},
 }
+var delegate = null;
```
Comment (NicholasSmilovic) [inline src/admob.ios.js]: I believe I was having, may still have, issues with weakly stored variables. This seems to fix it. The error was inconsistent so I'm not positive.
```
CONSOLE LOG file:///app/tns_modules/nativescript-admob/admob.js:249:16: GADRewardBasedVideoAdDelegateImpl initWithCallback
CONSOLE LOG file:///app/tns_modules/nativescript-admob/admob.js:256:16: calling loaded
***** Fatal JavaScript exception - application has been terminated. *****
Native stack trace:
1   0x103331c1f NativeScript::reportFatalErrorBeforeShutdown(JSC::ExecState*, JSC::Exception*, bool, bool)
2   0x10335b8ea NativeScript::FFICallback<NativeScript::ObjCMethodCallback>::ffiClosureCallback(ffi_cif*, void*, void**, void*)
3   0x103d28ced ffi_closure_unix64_inner
4   0x103d29642 ffi_closure_unix64
5   0x102a54ad9 GADNCanMakePaymentsSignalSource
6   0x102a54d77 GADNCanMakePaymentsSignalSource
7   0x10a6324e1 _dispatch_call_block_and_release
8   0x10a63354b _dispatch_client_callout
9   0x10a63f380 _dispatch_main_queue_callback_4CF
10  0x10a03e3e9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
11  0x10a038a76 __CFRunLoopRun
12  0x10a037e11 CFRunLoopRunSpecific
13  0x110aa81dd GSEventRunModal
14  0x10b58081d UIApplicationMain
15  0x103d2949d ffi_call_unix64
16  0x125ed9b80
JavaScript stack trace:
1   rewardBasedVideoAdDidReceiveAd@file:///app/tns_modules/nativescript-admob/admob.js:257:17
2   UIApplicationMain@[native code]
3   start@file:///app/tns_modules/tns-core-modules/application/application.js:272:26
4   run@file:///app/tns_modules/tns-core-modules/application/application.js:300:10
5   anonymous@file:///app/app.js:10:16
6   evaluate@[native code]
7   moduleEvaluation@[native code]
8   @[native code]
9   promiseReactionJob@[native code]
JavaScript error:
file:///app/tns_modules/nativescript-admob/admob.js:257:17: JS ERROR TypeError: this._loaded is not a function. (In 'this._loaded()', 'this._loaded' is undefined)
```

Comment (EddyVerbruggen) [reply src/admob.ios.js]: I have an idea, will change it slightly after merging.

**Comment by EddyVerbruggen** [comment]
This looks really good and I ❤️that you've also update the documentation and demo. I'll merge and test it.

**Comment by EddyVerbruggen** [comment]
Great work, thank you! Version 3.3.0 has now shipped to npm.

**Comment by JCenas** [comment]
Great job guys :)

**Pull request merged**